### DISCUSSION TITLE
Creating a Switch via NetBox

### DISCUSSION THREAD

**Discussion by remchuk** [netbox-community/netbox#13066] [Q&A] [answered]
Hey everyone!

So my team and I are making a flask server that will integrate any creations and changes in the NetBox into our devices.

I have ran into a issue though, when creating a switch in NetBox there is no way to assign IP to one of it's interfaces in the Creation section, however I want to make an automation for creating a switch.

Is there some way around it ? any suggestion ? Thanks a lot!

**Comment by candlerb** [comment]
See https://github.com/netbox-community/netbox/discussions/13026#discussioncomment-6294212

**Reply by candlerb** [reply]
Just to add: in Netbox a device has attributes "primary_ip4" and "primary_ip6". In order to set these, the user will have to add the IP address to an interface, and also mark it as the primary (this can be done in the same form).

If the automation you're referring to is downstream (your flask setup), then you'll use webhooks to get notified when a device is created or changed. When the device is created, it will have no primary_ip and so you can ignore it. However, when the user later sets a primary_ip, you'll get notified again, and then you can act on it.

However, if by "automation" you mean you want a user to go to a single web page and create a switch in one step (including setting its management IP address), you can write a [Custom Script](https://docs.netbox.dev/en/stable/customization/custom-scripts/) for this purpose.

Here's an example which is in a similar vein: it's a custom script to create a Virtual Machine + VM Interface + IP Address and set the VM's primary IP.
https://github.com/netbox-community/customizations/blob/master/scripts/create_vm.py

**Reply by remchuk** [reply]
Is there a way for a webhook to be sent only when status switched to active from planned for example ?

Or is there a way to send the change made when an update is triggered ?

Like a webhook is set and in the json sent it specifics what was changed?

**Reply by candlerb** [reply, marked as answer]
> Is there a way for a webhook to be sent only when status switched to active from planned for example ?

https://docs.netbox.dev/en/stable/integrations/webhooks/#conditional-webhooks

> in the json sent it specifics what was changed?

https://docs.netbox.dev/en/stable/integrations/webhooks/#default-request-body
(see snapshots.prechange and snapshots.postchange)

**Reply by remchuk** [reply]
Thank you very much !
